tcp-keepalive/internal/server/server.go

```diff
@@ -26,18 +29,31 @@ func initLogger() {
 	log = slog.New(slog.NewJSONHandler(os.Stdout, nil))
 }
 
-func NewServer(cfg *Config) (*Server, error) {
+func NewServer(cfg *Config, mcfg *metrics.Config) (*Server, error) {
 	if err := cfg.Validate(); err != nil {
 		return nil, err
 	}
 
-	log.Info("new server", slog.Any("config", cfg))
-	return &Server{Config: cfg}, nil
+	m, err := NewMetrics(mcfg)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info("new server", slog.Any("config", cfg), slog.Any("metrics", m))
+	return &Server{Config: cfg, metrics: m}, nil
 }
 
 func (s *Server) Run(ctx context.Context) error {
 	log = log.With("listen", s.ListenAddr)
 
+	if s.metrics.Export {
+		go func() {
+			for {
+				log.Error("serving metrics failed", slog.Any("error", s.metrics.Serve().Error()))
```

Comment:
I think it is better to split lines `s.metrics.Serve()` and log output.

Reply:
@terassyi Why do you think so? Is it for readability?

Reply:
It might not be good to include the server Serve() in a log output line.

Reply:
I split `s.metrics.Serve()` (also, the client too).